File: pythia-0.8/examples/tabulator/libtabulator/simpletab.f90
! -*- F90 -*-
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!
!                             Michael A.G. Aivazis
!                      California Institute of Technology
!                      (C) 1998-2005  All Rights Reserved
!
! <LicenseText>
!
! ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
!

subroutine simpletab(a, low, hi, step)

    implicit none
    double precision:: a, low, hi, step

    external exponential
    double precision:: exponential

    double precision:: x, r_exp

    call exponential_set(a)

    print '(1x,A15,A3,A15)', 'x       ', ' | ', 'value     '
    print '(1x,A33)', '----------------+----------------'

    do x = low, hi, step
        r_exp = exponential(x)
        print '(1x,F15.9,A3,F15.9)', x, ' | ', r_exp
    end do

end subroutine simpletab


! version
! $Id: simpletab.f90,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $

! End of file 

File: pythia-0.8/examples/tabulator/tests/simpletab.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                             Michael A.G. Aivazis
#                      California Institute of Technology
#                      (C) 1998-2005  All Rights Reserved
#
# <LicenseText>
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#

from pyre.applications.Script import Script


class SimpleApp(Script):


    class Inventory(Script.Inventory):

        import tabulator
        import pyre.inventory

        tabulator = pyre.inventory.facility("tabulator", factory=tabulator.simple)


    def main(self, *args, **kwds):
	simple = self.inventory.tabulator
        simple.tabulate()
            
        return


    def __init__(self):
        Script.__init__(self, "simpletab")
        return


# main

if __name__ == "__main__":
    app = SimpleApp()
    app.run()

# version
__id__ = "$Id: simpletab.py,v 1.1.1.1 2005/03/17 20:03:02 aivazis Exp $"

# End of file 
